src/subscript/ri_wellmod/ri_wellmod.py

```diff
@@ -0,0 +1,532 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+"""
+ ri_wellmod.py
+
+
+"""
+import argparse
+import os.path
+import fnmatch
+import shutil
+import xml.dom.minidom
+import re
+import logging
+from datetime import datetime
+
+import rips
+
+from subscript import getLogger
+
+DESCRIPTION = """
+``ri_wellmod`` is a utility to generate Eclipse well model definitions
+(WELSPECS/WELSPECL, COMPDAT/COMPDATL, WELSEGS, COMPSEGS) using ResInsight. The script
+takes as input a ResInsight project with wells and completions defined, in addition to
+an Eclipse case (either an initialized case or an input case with grid and PERMX|Y|Z
+and NTG defined in the GRDECL format).
+
+.. note:: Well names specified as command line arguments are assumed to refer to the
+   Eclipse well names, i.e., the completion export names as defined in the ResInsight
+   wells project.
+"""
+
+CATEGORY = "modelling.reservoir"
+
+EXAMPLES = """
+.. code-blck:: console
+
+ FORWARD_MODEL RI_WELLMOD(<RI_PROJECT>=<CONFIG_PATH>/../../resinsight/input/well_modelling/wells.rsp, # noqa: E501
+                          <ECLBASE>=<ECLBASE>,
+                          <OUTPUTFILE>=<RUNPATH>/eclipse/include/schedule/well_def.sch)
+
+ FORWARD_MODEL RI_WELLMOD(<RI_PROJECT>=<CONFIG_PATH>/../../resinsight/input/well_modelling/wells.rsp, # noqa: E501
+                          <ECLBASE>=<ECLBASE>,
+                          <OUTPUTFILE>=<RUNPATH>/eclipse/include/schedule/well_def.sch,
+                          <MSW>="A2;A4;'R*')
+
+ FORWARD_MODEL RI_WELLMOD(<RI_PROJECT>=<CONFIG_PATH>/../../resinsight/input/well_modelling/wells.rsp, # noqa: E501
+                          <ECLBASE>=<ECLBASE>,
+                          <OUTPUTFILE>=<RUNPATH>/eclipse/include/schedule/well_def.sch,
+                          <MSW>="A4",
+                          <XARG0>="--lgr",
+                          <XARG1>="A4:3;3;1")
+
+
+.. note:: More examples and options may be seen in the subscript docs for the script
+   ``ri_wellmod``, just replace ',' by ';' and note that spaces cannot be part of
+   argument strings, so you may need to use <XARGn> for the individual parts.
+
+"""
+
+
+logger = getLogger(__name__)
+
+RI_HOME = "/prog/ResInsight"
+DEFAULT_VERSION = "2020.10.1"
+
+
+def get_resinsight_exe(version):
+    """Return the path to the ResInsight install for a given version"""
+    return RI_HOME + "/resinsight_" + version + "_RHEL7/ResInsight"
+
```

Comment:
Would it not be more robust to check users path that hard coding a path? Maybe the subroutine could try to check path first and revert to a hard coded path if not found. This would open for the possibility of using test versions etc.

Reply:
I have ResInsight in `/usr/bin/ResInsight` on my Equinor developer laptop..

Reply:
Agreed - now first checking for 'ResInsight' using shutil.which, that should also make it possible to run the tests on gh (I think).